src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelGetDeclaredSymbolAPITests.cs

```diff
@@ -5267,6 +5267,380 @@ void F(int x = 2, = 3) { }
             Assert.Same(symbol1.ContainingSymbol, symbol2.ContainingSymbol);
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74348")]
+        public void ObjectInitializerIncompletePropertyValueDeclaration01()
+        {
+            var source = """
+                public class Thing
+                {
+                    public int Key { get; set; }
+                    public string? Value { get; set; }
+                }
+
+                public class Using
+                {
+                    public static Thing CreateThing(int key, string? value)
+                    {
+                        return new()
+                        {
+                            Key = key,
+                            Value,
+                        };
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var root = tree.GetCompilationUnitRoot();
+            var propertyDecls = root.DescendantNodes()
+                .OfType<ClassDeclarationSyntax>()
+                .First()
+                .ChildNodes()
+                .OfType<PropertyDeclarationSyntax>()
+                .ToArray();
+            var valuePropertyDecl = propertyDecls[1];
+            var valueProperty = model.GetDeclaredSymbol(valuePropertyDecl);
```

Comment:
>model.GetDeclaredSymbol(valuePropertyDecl) [](http://example.com/codeflow?start=32&length=42)

This is an extremely convoluted and inefficient way of getting a property symbol. It is much simpler to get it from symbol model. I think we have helpers that do that based on qualified name. "GetMemberSymbol" or something. That said, given the goal of this test, I don't think we even need to locate the symbol. See the next comment

Reply:
Done